persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -205,6 +208,16 @@ public static void populateEntities(Module.Builder moduleBuilder, SyntaxTree bal
                     if (fieldNode.typeName().kind().equals(SyntaxKind.ARRAY_TYPE_DESC)) {
                         type = ((ArrayTypeDescriptorNode) fieldNode.typeName()).memberTypeDesc();
                         fieldBuilder.setArrayType(true);
+                    } else if (fieldNode.typeName().kind().equals(SyntaxKind.OPTIONAL_TYPE_DESC)) {
+                        if (((OptionalTypeDescriptorNode) fieldNode.typeName()).typeDescriptor().kind()
```

Comment:
This seems to repeat multiple times. Let move the optional or array check to the top of the if else check. 
Also `SyntaxKind.RECORD_FIELD_WITH_DEFAULT_VALUE` is not supported now.

Reply:
this was updated in [bebd8e7](https://github.com/ballerina-platform/persist-tools/pull/145/commits/bebd8e733cc86a2a4b5b1b71e4c55f6040f00d41) and [4e4b08a](https://github.com/ballerina-platform/persist-tools/pull/145/commits/4e4b08a8c2ce8225a924f387836bedef241805d0)